src/Typeclasses/Classes/CommutativeMonoid.elm

```diff
@@ -0,0 +1,132 @@
+module Typeclasses.Classes.CommutativeMonoid exposing
```

Comment:
As I understand this module is merely about aliasing to `Monoid`. It will confuse people. Let's avoid that. If you feel like it, you can extend the docs on `Monoid` to say that it's commutative.

Reply:
Yes it does mostly alias Monoid.  There are some differences like list and string which are Monoids but not Commutative Monoids.

Reply:
What's the benefit to having it?

Reply:
Without the ability to enforce the commutative constraint the benefit is limited.  I was using it in the definition for Semiring and for segregating list and string to just monoid.  I agree that it may be verbose.

Reply:
We can redefine Semiring around just monoids.

Since CommutativeMonoid is just an alias to Monoid, it means that every instance of CommutativeMonoid is also an instance of Monoid and vice versa. This leaves CommutativeMonoid redundant, especially in the light of us having the ability to define multiple instances of Monoid per type, unlike in say Haskell.

Reply:
That's an excellent point!  I forgot how 'alias' is type permissive.  I can consolidate Monoid and CommutativeMonoid into Monoid and also Group and AbelianGroup into Group?  I suppose I could also do type CommutativeMonoid = CommutativeMonoid Monoid but that's still verbose?

Reply:
I agree. Let's avoid complication until we get a strong reason for it. After all we'll always be able to refactor this later.

So let's merge the CommutativeMonoid with Monoid and AbelianGroup with Group.